contracts/controller/src/healthfactor.rs

```diff
@@ -196,4 +255,160 @@ mod tests {
+        let (mut controller_contract, _token_address, user_account) =
+            init_price_volatility(0, 0, 10000, 100);
+
+        controller_contract.increase_supplies(
+            user_account.clone(),
+            AccountId::new_unchecked("dweth.near".to_string()),
+            WBalance::from(0),
+        );
+
+        controller_contract.increase_borrows(
+            user_account.clone(),
+            AccountId::new_unchecked("dweth.near".to_string()),
+            WBalance::from(0),
+        );
+
+        assert_eq!(
+            controller_contract.get_health_factor(user_account.clone()),
+            controller_contract.get_health_threshold(),
+            "Test for account w/o collaterals and borrows has been failed"
+        );
+    }
+
+    #[test]
+    fn test_health_factor_with_supply() {
+        let (mut controller_contract, _token_address, user_account) =
+            init_price_volatility(0, 0, 10000, 100);
+
+        controller_contract.increase_supplies(
+            user_account.clone(),
+            AccountId::new_unchecked("dweth.near".to_string()),
+            WBalance::from(100),
+        );
+
+        controller_contract.increase_borrows(
+            user_account.clone(),
+            AccountId::new_unchecked("dweth.near".to_string()),
+            WBalance::from(0),
+        );
+
+        // Ratio that represents 150%
+        assert_eq!(controller_contract.get_health_factor(user_account), 15000);
+    }
+
+    #[test]
+    fn test_health_factor_with_supply_and_borrow_scenario_1() {
+        let (mut controller_contract, _token_address, user_account) =
+            init_price_volatility(0, 0, 10000, 100);
+
+        controller_contract.increase_supplies(
+            user_account.clone(),
+            AccountId::new_unchecked("dweth.near".to_string()),
+            WBalance::from(100),
+        );
+
+        controller_contract.increase_borrows(
+            user_account.clone(),
+            AccountId::new_unchecked("dweth.near".to_string()),
+            WBalance::from(70),
+        );
+
+        // Ratio that represents 142.85%
+        assert_eq!(controller_contract.get_health_factor(user_account), 14285);
+    }
+
+    #[test]
+    fn test_health_factor_with_supply_and_borrow_scenario_2() {
+        let (mut controller_contract, _token_address, user_account) =
+            init_price_volatility(10000, 100, 10000, 100);
+
+        controller_contract.increase_supplies(
+            user_account.clone(),
+            AccountId::new_unchecked("dweth.near".to_string()),
+            WBalance::from(200),
+        );
+
+        controller_contract.increase_borrows(
+            user_account.clone(),
+            AccountId::new_unchecked("dwnear.near".to_string()),
+            WBalance::from(100),
+        );
+
+        // Ratio that represents 200%
+        assert_eq!(controller_contract.get_health_factor(user_account), 20000);
+    }
+
+    #[test]
+    fn test_health_factor_with_supply_and_borrow_scenario_3() {
```

Comment:
@dmytro-shash 

Scenario 5: 
  Given supply is 200 WETH
  And borrow is 100 WNEAR
  And Price is {ticker_id: WETH, value: 10000, volatility: 100%, fraction_digits: 4}
  And Price is {ticker_id: WNEAR, value: 10000, volatility: 100%, fraction_digits: 4}
  Then health_factor equal 200%
  When Price is {ticker_id: WETH, value: 5000, volatility: 100%, fraction_digits: 4}
  And Price is {ticker_id: WNEAR, value: 20000, volatility: 100%, fraction_digits: 4}
  Then health_factor equal 50%